pkg/cmd/workflow.go

```diff
@@ -0,0 +1,364 @@
+
+	if cerr != nil {
+		return cerr
+	}
+
+	if !proceed {
+		return nil
+	}
+
+	Dialog.Progress("Deleting workflow...")
+
+	_, err := Client.DeleteWorkflow(workflowName)
+
+	if err != nil {
+		return err
+	}
+
+	Dialog.Info("Workflow successfully deleted")
+
+	return nil
+}
+
+func newDeleteWorkflowCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "delete [workflow name]",
+		Short: "Delete a relay workflow",
+		Args:  cobra.MaximumNArgs(1),
+		RunE:  doDeleteWorkflow,
+	}
+
+	return cmd
+}
+
+func parseParameter(str string) (key, value string) {
+	idx := strings.Index(str, "=")
+
+	// TODO: This behavior will basically silently discard parameters that are
+	// in the wrong format. Should this, like, panic perhaps? Or notify the user that their
+	// parameters are specified incorrectly?
+	if idx < 0 {
+		return
+	}
+
+	key = str[0:idx]
+	value = str[idx+1:]
+	return
+}
+
+func parseParameters(strs []string) map[string]string {
+	res := make(map[string]string)
+
+	for _, str := range strs {
+		key, val := parseParameter(str)
+
+		// value of empty string could, indeed, be a valid parameter.
+		if key != "" {
+			res[key] = val
+		}
+	}
+
+	return res
+}
+
+func doRunWorkflow(cmd *cobra.Command, args []string) error {
+	params, err := cmd.Flags().GetStringArray("parameter")
+
+	if err != nil {
+		panic(err)
+	}
+
+	// TODO: Same here as above. Could really DRY all this up.
+	name, err := getWorkflowName(args)
+
+	if err != nil {
+		return err
+	}
+
+	Dialog.Progress("Starting your workflow...")
+
+	resp, err := Client.RunWorkflow(name, parseParameters(params))
+
+	if err != nil {
+		// TODO: This error should be translated for the user. Right now it just
+		// says whatever the service says.
+		return err
+	}
+
+	link := format.GuiLink(Config, "/workflows/ec2-reaper/runs/%d/graph", resp.Run.RunNumber)
+	Dialog.Info(fmt.Sprintf("Your run has started. Monitor it's progress: %s", link))
+
+	return nil
+}
+
+func newRunWorkflowCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "run [workflow name]",
+		Short: "Invoke a relay workflow",
+		Args:  cobra.MaximumNArgs(1),
+		RunE:  doRunWorkflow,
+	}
+
+	return cmd
+}
+
+func doDownloadWorkflow(cmd *cobra.Command, args []string) error {
+	name, err := getWorkflowName(args)
+
+	if err != nil {
+		return err
+	}
+
+	body, err := Client.DownloadWorkflow(name)
+
+	if err != nil {
+		// TODO: This error should be translated for the user. Right now it just
+		// says whatever the service says.
+		return err
+	}
+
+	// Just plop the whole thing out output.
+	Dialog.WriteString(body)
+
+	return nil
+}
+
+func newDownloadWorkflowCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "download [workflow name]",
+		Short: "Download a workflow from the service.",
+		Args:  cobra.MaximumNArgs(1),
+		RunE:  doDownloadWorkflow,
+	}
+
+	return cmd
+}
+
+func doListWorkflows(cmd *cobra.Command, args []string) error {
+	resp, err := Client.ListWorkflows()
+
+	if err != nil {
+		panic(err)
```

Comment:
see above

Reply:
this actually shouldn't have been a panic in the first place...